tutorials/zkSync-RedStone-stable-price-marketplace-tutorial/README.md

```diff
@@ -0,0 +1,402 @@
+  const marketplace = await getContractInstance("marketplace");
+
+  // Wrapping marketplace contract instance.
+  // It enables fetching data from redstone data pool
+  // for each contract function call
+  try {
+    const wrappedMarketplaceContract = WrapperBuilder.wrap(
+      marketplace
+    ).usingDataService(
+      {
+        dataServiceId: "redstone-main-demo",
+        uniqueSignersCount: 1,
+        dataFeeds: ["ETH"],
+      },
+    );
+
+    // Checking expected amount
+    const expectedEthAmount = await wrappedMarketplaceContract.getPrice(orderId);
+
+    // Sending buy tx
+    const buyTx = await wrappedMarketplaceContract.wi(orderId, {
+      value: expectedEthAmount.mul(101).div(100), // a buffer for price movements
+    });
+    await buyTx.wait();
+
+    return buyTx;
+  } catch { 
+    const errText = "Error happened while buying the NFT";
+    alert(errText);
+  }
+}
+```
+
+### Test dApp locally
+
+#### 1. Check if tests pass
+```sh
+yarn test
+``` 
+
+#### 2. Compile contracts
+```sh
+yarn compile
+``` 
+
+#### 3. Deploy contracts on local blockchain
+You need to populate .env file with private key for deployment e.g.
+```
+WALLET_PRIVATE_KEY=0x7726827caac94a7f9e1b160f7ea819f172f7b6f9d2a97f992c38edeab82d4110
+```
+and then run
+```sh
+yarn deploy:local
+```
+
+#### 4. Run react app
+```sh
+yarn app:start
+```
+
+The app should be running on http://localhost:3000
+
+#### 3. Configure metamask
+
+##### 3.1 Add local hardhat network to metamask
+Select `Networks dropdown` -> `Add network` and enter the following details:
+**Network Name**|**hardhat-local**
+:-----:|:-----:
+New RPC URL|http://localhost:3050
+Chain ID|270
+Currency Symbol|ETH
+
+Then hit the `Save` button.
+
+##### 3.2 Add local wallets to metamask
+- `User 1`: `0xac1e735be8536c6534bb4f17f06f6afc73b2b5ba84ac2cfb12f7461b20c0bbe3`
+- `User 2`: `0xd293c684d884d56f8d6abd64fc76757d3664904e309a0645baf8522ab6366d9e`
+
+
+#### 4. Explore the app in browser
+
+<img alt="stable-marketplace-app" src="docs/img/stable-marketplace-app.png" width="800" />
+
+##### Mint NFTs
+After visiting the app first time you will see an almost empty screen with the `+ Mint new NFT` link. Click this link to mint new NFTs. After the minting transaction confirmation you will see your NFT in the left column.
+
+<img alt="my-nfts" src="docs/img/my-nfts.png" width="300" />
+
+##### Post sell orders
+Once you mint any NFTs, you can post sell order for each one of them. Click the SELL button and provide the USD value. You will be asked to confirm 2 transactions: for NFT transfer approval, and for the marketplace order creation. After their confirmation, you will see your order in the Orders column.
+
+<img alt="orders" src="docs/img/orders.png" width="300" />
+
+##### Buy NFTs
+You can also switch metamask account and buy the NFT. I would recommend to open the developer tools in browser at the network tab and explore network requests that are being sent before the buy transaction sending.
```

Comment:
Sadly I try and get the following error:

<img width="1420" alt="image" src="https://github.com/zkSync-Community-Hub/tutorials/assets/10233439/ba5cfbca-77ed-4ffc-8356-87f33350c535">

Only one request is sent:

<img width="555" alt="image" src="https://github.com/zkSync-Community-Hub/tutorials/assets/10233439/00afa639-176a-42d8-b1eb-9a4aacddce06">

Reply:
Fixed, the problem was with an invalid function name in the tutorial text